### DISCUSSION TITLE
OTBR device setup steps for RPi 4 hardware

### DISCUSSION THREAD

**Discussion by nikhil30081995** [openthread/openthread#8419] [Q&A]
Hello, 
 I have a product developed using nRF5 SDK for Thread with certified OTBR using RPi 3B+ using nRF52840 NCP device. We were currently using OTBR image provided by Nordic itself in the SDK. 

https://openthread.io/certification/border-router#thread_group_membership

Now we are looking to migrate our Border Router hardware to RPi 4 version. So i would like to know if the following steps given in the below link are still valid for OTBR setup on RPI4 ? 

https://openthread.io/certification/border-router/device-setup

**Comment by nikhil30081995** [comment]
Following steps from doc https://openthread.io/certification/border-router/device-setup does not install wpantund.  

I have performed sub step 1,2,3 of Step 4 but in sub step 5, when i try to open **wpantund.conf** there is no such file available.

**Reply by jwhui** [reply]
The current main branch of https://github.com/openthread/ot-br-posix/ does not support `wpantund`.

**Comment by jwhui** [comment]
Unfortunately, the border router certification guide in https://openthread.io/certification/border-router/device-setup has not been updated to support the latest code base.

**Reply by nikhil30081995** [reply]
> Unfortunately, the border router certification guide in https://openthread.io/certification/border-router/device-setup has not been updated to support the latest code base.

As my product is developed using OTBR based on wpantund and NCP will the below link still work for developing
the OTBR image for RPi 4. The doc currently shows the setup for RPi3B+ 

https://openthread.io/certification/border-router/device-setup

**Reply by jwhui** [reply]
I cannot really say as I have not tried running through the guide on a RPi 4.

**Reply by nikhil30081995** [reply]
> I cannot really say as I have not tried running through the guide on a RPi 4.

okay thanks for info. I will check that out. 

But before performing the steps given in doc, i have to install wpantund right ? Because the step's doesn't install wpantund by default .

**Reply by jwhui** [reply]
The [Setup OpenThread Border Router](https://openthread.io/certification/border-router/device-setup#set-up-openthread-border-router) step should install `wpantund`.

**Reply by nikhil30081995** [reply]
> The [Setup OpenThread Border Router](https://openthread.io/certification/border-router/device-setup#set-up-openthread-border-router) step should install `wpantund`.

Hello, the below step doesn't install wpantund in Rpi, as the configuration file is not formed in /etc folder. 

Step 3: Compile and install OTBR and wpantund, using the OTBR certification commit for wpantund: 
WPANTUND_COMMIT=8c189c6 ./script/setup

**Reply by jwhui** [reply]
The setup script does install `wpantund`. See: https://github.com/openthread/ot-br-posix/blob/72ed754c7230fc45dbacbe5a59b3b6ed87333c66/script/setup#L63

**Reply by nikhil30081995** [reply]
> The setup script does install `wpantund`. See: https://github.com/openthread/ot-br-posix/blob/72ed754c7230fc45dbacbe5a59b3b6ed87333c66/script/setup#L63

As given in docs, if i checkout the branch, my head is not moved to any specific location.  
git checkout -b thread-br-certified-20180819 

And below is the setup code snippet : 
`main()
{
    # shellcheck source=/dev/null
    . "$BEFORE_HOOK"
    extend_sudo_timeout
    setup_swapfile
    disable_services
    otbr_uninstall
    border_routing_uninstall
    network_manager_uninstall
    dhcpv6_pd_uninstall
    nat64_uninstall
    dns64_uninstall
    rt_tables_uninstall
    ipforward_uninstall
    firewall_uninstall

    firewall_install
    ipforward_install
    rt_tables_install
    nat64_install
    dns64_install
    network_manager_install
    dhcpv6_pd_install
    border_routing_install
    otbr_install
    # shellcheck source=/dev/null
    . "$AFTER_HOOK"
}
`

Can you tell me which branch is used for this setup : https://openthread.io/certification/border-router/device-setup#set-up-openthread-border-router

**Reply by jwhui** [reply]
Try removing the `-b` argument:
```
git checkout thread-br-certified-20180819
```

**Reply by nikhil30081995** [reply]
> Try removing the `-b` argument:
> 
> ```
> git checkout thread-br-certified-20180819
> ```

When i ran the the "WPANTUND_COMMIT=8c189c6 ./script/setup", my vnc connection went off and a Border Router Access point got generated. I tried to connect to the BorderRouter-AP with password "12345678" but it says unable to connect. 

I gained access of RPi using UART, and tried to connect to a wifi network directly using nmcli. But for some reason my i am unable to scan the wifi network even when the radio status is enabled. 


I have flashed the raspbian image again on RPi 4. I would like to know the hardware setup as well before re-running the setup script.

Current Hardware Setup: RPI4 connected to internet using ethernet cable.

**Reply by jwhui** [reply]
I recall some issues with network manager and soft AP in later Raspberry Pi OS releases. I'm not sure if that was ever resolved. I would just avoid using soft AP.

**Reply by nikhil30081995** [reply]
> I recall some issues with network manager and soft AP in later Raspberry Pi OS releases. I'm not sure if that was ever resolved. I would just avoid using soft AP.

Can you tell how can i skip the soft AP installation or configuration when setting up the border router.

**Reply by jwhui** [reply]
Set `NETWORK_MANAGER=0`. See https://github.com/openthread/ot-br-posix/blob/72ed754c7230fc45dbacbe5a59b3b6ed87333c66/script/_network_manager#L326

**Reply by nikhil30081995** [reply]
> Set `NETWORK_MANAGER=0`. See https://github.com/openthread/ot-br-posix/blob/72ed754c7230fc45dbacbe5a59b3b6ed87333c66/script/_network_manager#L326

Okay thanks for information. Will set **NETWORK_MANAGER=0,** in ot-br-posix/examples/raspbian/default file and retry border router setup.

Currently my /etc/dhcpcd.conf file is modified for accessing rpi on static ip address over eth0 interface. 

While installing the dependencies, it asked me whether to modify /etc/dhcpcd.conf file or not ? By default the answer was no. Do you have any idea, should i allow the /etc/dhcpcd.conf file to be modified ? 

![image](https://user-images.githubusercontent.com/84308431/205553879-a96ace8d-46b8-4bee-a781-40e515fa7b7f.png)

**Reply by jwhui** [reply]
Since you modified `/etc/dhcpcd.conf` with your own custom configs, I would show the differences and decide what to do from there.

**Reply by nikhil30081995** [reply]
> Since you modified `/etc/dhcpcd.conf` with your own custom configs, I would show the differences and decide what to do from there.

Okay thanks for information.
I would like to know open thread use which type of modulation ?

**Reply by jwhui** [reply]
> I would like to know open thread use which type of modulation ?

IEEE 802.15.4 2.4 GHz O-QPSK.

**Reply by nikhil30081995** [reply]
> > I would like to know open thread use which type of modulation ?
> 
> IEEE 802.15.4 2.4 GHz O-QPSK.

Thanks for information. 
Currently in my existing Border Router setup on RPi3B+ based on Nordic OpenThread Border Router image, the dhcpcd.conf file looks as shown in below. 

**/etc/dhcpcd.conf file**
`
nohook lookup-hostname
nohook wpa_supplicant

denyinterfaces nat*

noipv6rs

interface wpan0
nodhcp
nodhcp6

interface eth0
iaid 1
ipv6rs
ia_na 2
ia_pd 3/::/63 wpan0/1

interface wlan0
iaid 4
ipv6rs
ia_na 5
ia_pd 6/::/63 wpan0/1
`


I would like to know can i modify the dhcpcd.conf file now, and provide it static ip address for wifi interface ? Will it anyhow effect the wpan0 interface ? 

I would like to try something like this :
interface wlan0
#iaid 4
#ipv6rs
#ia_na 5
#ia_pd 6/::/63 wpan0/1
static ip_address=192.168.0.200
static routers=192.168.0.1
static domain_name_servers=192.168.0.1 8.8.8.8

**Reply by jwhui** [reply]
As long as you scope the DHCP client config to `wlan0`, then it should not affect `wpan0` interface.

**Reply by nikhil30081995** [reply]
> As long as you scope the DHCP client config to `wlan0`, then it should not affect `wpan0` interface.

Hello @jwhui ,
Thanks for assistance. 

In my existing OTBR setup was, the design which was used was of NCP type because the example are based on MQTT-SN publisher example from nRF5 SDK for Thread. The docs of example were based on NCP design. 

Now as i am looking to change the OTBR design i.e from NCP to RCP design will it any way effect the example based on nRF5 SDK for Thread working in future ?

**Reply by jwhui** [reply]
If MQTT-SN end point is operating as a Linux process binding to a standard network socket, then there should be little effect whether it is NCP or RCP. Both models present Thread as a standard network interface in Linux.

**Reply by nikhilkomalan** [reply]
Hey @jwhui ,
I was able to setup OTBR using nRF52840 RCP design. I have also setup the MQTT-SN gateway to make it compatible with my examples. 

Now after few minutes of proper working of my thread network, some of my devices get disconnected from the Thread Network i.e it's state changed to state 1. I have attached the log file i read from the OTBR  "journactctf -ef " command. 
Can you please have a look and tell me why my devices are getting disconnected from thread network  ? 

Log File: 

[ot_log.txt](https://github.com/openthread/openthread/files/10251450/ot_log.txt)

**Reply by jwhui** [reply]
See https://github.com/openthread/openthread/discussions/8556#discussioncomment-4454358
